doc/source/notebooks/advanced/varying_noise.pct.py

```diff
@@ -200,14 +206,8 @@ def generate_data(N=100):
 def objective_closure():
     return - model.log_marginal_likelihood()
 
-from gpflow.utilities import set_trainable
-set_trainable(model.kernel, False)
-set_trainable(model.likelihood, False)
-
-opt = gpflow.optimizers.Scipy()
-opt.minimize(objective_closure,
-             model.trainable_variables,
-             options=dict(maxiter=ci_niter(1000)))
+for _ in range(ci_niter(1000)):
```

Comment:
why aren't we interleaving the optimisation of natgrad with scipy here?

Reply:
cos only q_mu and q_sqrt were left as trainable, so I guess there's nothing else to train

see
```
set_trainable(model.kernel, False)
set_trainable(model.likelihood, False)
```
in original

Reply:
Here we are explicitly only optimising variational parameters, not hyperparameters (see line 213/214)